File: server/app/models/animal.py
from typing import Optional
from pydantic import BaseModel


class Position(BaseModel):
    x: int
    y: int
    width: int
    height: int


class Animal(BaseModel):
    animal: str
    confidence: float
    position: Position


class AnimalResponse(BaseModel):
    success: bool
    message:str
    data: Optional[Animal] = None

File: client/src/types/animal.ts
export interface Position {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface AnimalData {
  animal: string;
  confidence: number;
  position: Position;
}

export interface AnimalResponse {
  success: boolean;
  message: string;
  data: AnimalData | null;
}
